src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -57,39 +74,49 @@ void setUserStore(UserStore userStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
+
     String requestUrl = request.getRequestURI();
     String username = requestUrl.substring("/users/".length());
 
     User user = userStore.getUser(username);
+    if (user == null){
+      //redirects user to the login page if user is not logged in
+      response.sendRedirect("/login");
+      return;
+    }
+
+    UUID userID = UserStore.getInstance().getUser(username).getId();
+    if (userID == null){
+      //redirects to login page if there is no user id
+      response.sendRedirect("/login");
+      return;
+    }
```

Comment:
This might not be a big deal but: `UserStore.getIstance()` should be replaced with `userStore`, the instance you already have, this is because when testing you'll provide a mock UserStore instance.
Or even better, Why not using `user.getId()`? you already have a non-null user

Reply:
Sounds like a good idea to use user.getId() directly.